compiler/rustc_middle/src/ty/context.rs

```diff
@@ -92,6 +91,70 @@ pub trait OnDiskCache<'tcx>: rustc_data_structures::sync::Sync {
     fn serialize(&self, tcx: TyCtxt<'tcx>, encoder: &mut FileEncoder) -> FileEncodeResult;
 }
 
+pub struct TyInterner<'tcx> {
+    pub tcx: TyCtxt<'tcx>,
+}
+
+/// We don't ever actually need this. It's only required for derives.
+impl<'tcx> Hash for TyInterner<'tcx> {
+    fn hash<H: Hasher>(&self, _state: &mut H) {}
```

Comment:
Maybe panic in all these impls to make sure we don't actually use them

Reply:
I was not a fan of having these impls at all actually, so I just ended up removing these in favor of including (an edited version) of the derive macro expansions for the common traits on `TyKind`.